zk-collectd.py

```diff
@@ -99,30 +125,24 @@ def _parse_line(self, line):
 
 def read_callback():
     """Get stats for all the servers in the cluster."""
+    stats = {}
     for conf in CONFIGS:
         for host in conf['hosts']:
-            try:
-                zk = ZooKeeperServer(host, conf['port'])
-                stats = zk.get_stats()
-                for k, v in stats.items():
-                    try:
-                        val = collectd.Values(plugin='zookeeper',
-                                              meta={'0': True})
-                        val.type = 'counter' if k in COUNTERS else 'gauge'
-                        val.type_instance = k
-                        val.values = [v]
-                        val.plugin_instance = conf['instance']
-                        val.dispatch()
-                    except (TypeError, ValueError):
-                        collectd.error(('error dispatching stat; host=%s, '
-                                        'key=%s, val=%s') % (host, k, v))
-                        pass
-            except socket.error:
```

Comment:
Is it safe to let this error bubble up? Doesn't collectd automatically disable a plugin if it errors out too much?